lib/mutant/ast/meta.rb

```diff
@@ -81,7 +44,8 @@ def assignment?
         # @api private
         #
         def attribute_assignment?
-          arguments.one? && attribute_assignment_selector?
+          !Types::METHOD_OPERATORS.include?(selector) &&
+          selector.to_s.end_with?(ATTRIBUTE_ASSIGNMENT_SELECTOR_SUFFIX)
```

Comment:
All these guards are needed to not mutate code like:

```
foo.bar = baz
```

```
(send (send nil :foo) :bar= (send nil :baz))
```

into an (invalid) AST that would unparse (unparser trusts the AST 100%) into:

```
bar = baz
```
(send nil :bar= (send nil :baz))

The AST in this case would say call `#bar=` on implicit self. But the ruby parser would say: Assign the lvar `bar=`. I see how this could be a cool mutation, but if we do this: It should be explicit, and not an artifact of a transform to invalid AST.

Valid AST would, for this case be:

```
(lvasgn :bar (send nil :baz))
```